### PR TITLE
Add semantic color tokens to BaseStyles

### PR DESCRIPTION
**Pull request by robinmetral**
## Purpose

This PR adds semantic color tokens to the Circuit UI BaseStyles.

The variables and color values are defined in Figma ([internal ref](https://www.figma.com/file/OgPQeoNZ2QoY7hZvy0ybk2/%F0%9F%8C%88-COLOR-TOKENS?node-id=975%3A4211&t=TvIKgWoOpmAzC4sF-4)).

## Approach and changes

- add the variables to the `:root` pseudo-class in `BaseStyles` (injected as global styles by Emotion)
- listed the variables on the Storybook [theme documentation](https://oss-circuit-ui-git-semantic-color-tokens.sumup-vercel.app/?path=/docs/features-theme--page)

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* ~Unit and integration tests~
* [x] Meets minimum browser support (uses CSS custom properties)
* ~Meets accessibility requirements~

### CODE REVIEW

**Comment by robinmetral** [comment]
(Apologies for the noise—opened this early by mistake)

File: packages/circuit-ui/components/BaseStyles/BaseStylesService.ts
Comment (connor-baer) [inline packages/circuit-ui/components/BaseStyles/BaseStylesService.ts:47]: CSS comments should use a different syntax:

```suggestion
    /* Normal backgrounds */
```

Comment (robinmetral) [reply packages/circuit-ui/components/BaseStyles/BaseStylesService.ts:47]: 🙏 Yes. I'll remove these anyways before merging (I think the token names are pretty self-explanatory)

Comment (connor-baer) [reply packages/circuit-ui/components/BaseStyles/BaseStylesService.ts:47]: I think they’re nice for grouping :)

File: packages/circuit-ui/components/BaseStyles/BaseStylesService.ts
Comment (connor-baer) [inline packages/circuit-ui/components/BaseStyles/BaseStylesService.ts:62]: How will users perceive/differentiate these different states?

Comment (robinmetral) [reply packages/circuit-ui/components/BaseStyles/BaseStylesService.ts:62]: In this case it looks like there is no differentiation. This could for instance be used in the ODL black buttons. From a standards perspective (WCAG 2.1 AA), nothing states that hovered or pressed states should be visually different from a UI element's idle state.

So we fall back to a design question—I'm happy to raise this with design again in the context of ODL (which introduces a lot of these black and white styles) but in the meantime I'd suggest moving forward here (no blocker and this is a design decision).

Comment (connor-baer) [reply packages/circuit-ui/components/BaseStyles/BaseStylesService.ts:62]: I'm surprised this isn't an accessibility failure. I would at least classify it as a usability issue. Do you know if there's any other indicator planned in the new button design? 

I agree that this shouldn't block this PR, but I'd like to question this decision and improve it in the future.

**Review by connor-baer — APPROVED** [review]

**Comment by robinmetral** [comment]
@connor-baer re-requesting a quick review for the documentation ([preview](https://oss-circuit-ui-git-semantic-color-tokens.sumup-vercel.app/?path=/docs/features-theme--page)). I plan on expanding on it in the future—for now the colors are simply listed on the Theme page, and there's a convenient button to copy the custom property name.

**Review by connor-baer — APPROVED** [review]
Amazing 🎨

**Pull request merged**
